pkgs/development/libraries/ffmpeg/generic.nix

```diff
@@ -0,0 +1,527 @@
+    #(mkFlag (zvbi != null) "2.1" "libzvbi")
+    /*
+     * Developer flags
+     */
+    (mkFlag debugDeveloper "0.5" "debug")
+    (mkFlag optimizationsDeveloper "0.5" "optimizations")
+    (mkFlag extra-warningsDeveloper "0.5" "extra-warnings")
+    (mkFlag strippingDeveloper "0.5" "stripping")
+    
+    # Disable mmx support for 0.6.90
+    (verFix null "0.6.90" "--disable-mmx")
+  ];
+
+  nativeBuildInputs = [ perl pkgconfig texinfo yasm ];
+
+  buildInputs = [
+    alsaLib bzip2 celt faac faad2 fdk_aac fontconfig freetype frei0r fribidi
+    game-music-emu gnutls gsm jack2 ladspaH lame libass libbluray libbs2b
+    libcaca libdc1394 libmodplug libogg libopus libraw1394 libssh libtheora
+    libva libvdpau libvpx libvorbis libwebp libX11 libxcb libXext libXfixes
+    libXv lzma mesa openal opencv openjpeg_1 openssl pulseaudio rtmpdump
+    samba SDL soxr speex v4l_utils vid-stab wavpack x264 x265 xavs xvidcore
+    zeromq4 zlib
+  ];
+
+  # Build qt-faststart executable
+  buildPhase = optional (qt-faststartProgram && (reqMin "0.9")) ''make tools/qt-faststart'';
+  postInstall = optional (qt-faststartProgram && (reqMin "0.9")) ''cp -a tools/qt-faststart $out/bin/'';
+
+  enableParallelBuilding = true;
+
+  /* Cross-compilation is untested, consider this an outline, more work
+     needs to be done to portions of the build to get it to work correctly */
+  crossAttrs = let
+    os = ''
+      if [ "${stdenv.cross.config}" = "*cygwin*" ] ; then
+        # Probably should look for mingw too
+        echo "cygwin"
+      elif [ "${stdenv.cross.config}" = "*darwin*" ] ; then
+        echo "darwin"
+      elif [ "${stdenv.cross.config}" = "*freebsd*" ] ; then
+        echo "freebsd"
+      elif [ "${stdenv.cross.config}" = "*linux*" ] ; then
+        echo "linux"
+      elif [ "${stdenv.cross.config}" = "*netbsd*" ] ; then
+        echo "netbsd"
+      elif [ "${stdenv.cross.config}" = "*openbsd*" ] ; then
+        echo "openbsd"
+      fi
+    '';
+  in {
+    dontSetConfigureCross = true;
+    configureFlags = configureFlags ++ [
+      "--cross-prefix=${stdenv.cross.config}-"
+      "--enable-cross-compile"
+      "--target_os=${os}"
+      "--arch=${stdenv.cross.arch}"
+    ];
+  };
+
+  /* TODO: In the future more FFmpeg optionals should be added so that packages that
+     depend on FFmpeg can check to make sure a required feature is enabled.  Since
+     features are version dependent, versioning needs to be handled as well */
+  passthru = {
+    vdpauSupport = libvdpau != null;
+  };
+
+  meta = {
+    description = "A complete, cross-platform solution to record, convert and stream audio and video";
+    homepage = http://www.ffmpeg.org/;
+    longDescription = ''
+      FFmpeg is the leading multimedia framework, able to decode, encode, transcode, 
+      mux, demux, stream, filter and play pretty much anything that humans and machines 
+      have created. It s the most obscure ancient formats up to the cutting edge. 
```

Comment:
Yeah, Thanks, I'll change it when I update the commit to remove the extra branches.